test/test_images/latencymako_sender_receiver/main.go

```diff
@@ -254,33 +264,28 @@ func main() {
 		vegeta.MaxWorkers(workers),
 	).Attack(targeter, combinedPacer, totalBenchmarkDuration, defaultEventType+"-attack")
 
-	go processVegetaResult(vegetaResults)
-
-	// count errors
-	var publishErrorCount int
-	var deliverErrorCount int
-	for eventState := range resultCh {
-		switch eventState {
-		case dropped:
-			deliverErrorCount++
-		case undelivered:
-			publishErrorCount++
-		}
-	}
+	// doneCh is closed as soon as all results have been processed
+	doneCh := make(chan struct{})
+	go processVegetaResult(vegetaResults, doneCh)
+	<-doneCh
 
-	printf("Publish error count: %v", publishErrorCount)
-	printf("Deliver error count: %v", deliverErrorCount)
+	printf("---- END BENCHMARK ----")
 
-	printf("--- END BENCHMARK ---")
+	printf("Sending collected data to the aggregator")
 
-	// publish error counts as aggregate metrics
-	q.AddRunAggregate("pe", float64(publishErrorCount))
-	q.AddRunAggregate("de", float64(deliverErrorCount))
+	printf("%-15s: %d", "Sent count", len(sentEvents.Events))
+	printf("%-15s: %d", "Accepted count", len(acceptedEvents.Events))
+	printf("%-15s: %d", "Failed count", len(failedEvents.Events))
+	printf("%-15s: %d", "Received count", len(receivedEvents.Events))
 
-	out, err := q.Store()
-	if err != nil {
-		fatalf("q.Store error: %v: %v", out, err)
-	}
+	// Create a connection to the aggregator
+	c, connclose := aggregatorClient()
+	defer connclose()
+
+	sendEventsRecord(c, &sentEvents)
```

Comment:
Do we expect the clients to always send all 4 of these requests? If so, maybe they can be combined into a single request. Then `--expect-records` is equal to the pod count instead of pods*4.